core/azure-core/src/test/java/com/azure/core/http/policy/HostPolicyTests.java

```diff
@@ -32,17 +32,19 @@ public void withPort() throws MalformedURLException {
     }
 
     private static HttpPipeline createPipeline(String host, String expectedUrl) {
-        return new HttpPipeline(new MockHttpClient() {
-            @Override
-            public Mono<HttpResponse> send(HttpRequest request) {
-                return Mono.empty(); // NOP
-            }
-        },
-        new HostPolicy(host),
-            (context, next) -> {
+        return HttpPipeline.builder()
+            .httpClient(new MockHttpClient() {
+                @Override
+                public Mono<HttpResponse> send(HttpRequest request) {
+                    return Mono.empty(); // NOP
+                }
+            })
```

Comment:
It would be nice to work through this code and replace the anonymous inner classes with lambdas.

Reply:
I made the MockHttpClient non-abstract and overrode the send method to return a no-op.

Reply:
Issue #3674 written to track the cleanup of these anonymous inner classes.